### ISSUE TITLE
named from munin-node on fedora33 fails

### ISSUE THREAD

**Issue by dwreski** [munin-monitoring/contrib#1153]
Hi, the /usr/share/munin/plugins/named included with munin-node in fedora33 doesn't work with bind-9.11.24.

In the named.stats file, no such variable as XSTATS or NSTATS exists in the named.stats file so the whole script fails.

**Comment by sumpfralle** [comment]
Did you follow the hints in the [documentation of that plugin](https://gallery.munin-monitoring.org/plugins/munin/named/)?

The plugin requires a change of the bind configuration to work.

**Comment by dwreski** [comment]
Yes, I should have been more clear about that in my initial post. I have updated /etc/named.conf to include "statistics-interval 1;". 

```
[named]
group named
env.logfile /var/named/data/named.stats
```
```
# ls -l  /var/named/data/named.stats
-rw-r--r-- 1 named named 22721802 Dec  6 19:41 /var/named/data/named.stats
```

named.stats attached here.
[named.stats.txt](https://github.com/munin-monitoring/contrib/files/5649882/named.stats.txt)

**Comment by sumpfralle** [comment]
> I have updated /etc/named.conf to include "statistics-interval 1;".

OK, that's good.

Regarding `/var/named/data/named.stats`: maybe you were somehow mislead, that this file would be relevant?
After taking a look at the [plugin code](https://gallery.munin-monitoring.org/plugins/munin/named/), I can only see references for `/var/log/daemon.log`. Or maybe the named log entries end up in a different file on your system?

**Comment by dwreski** [comment]
That may be the default name on debian, but on fedora, it's named.stats, or whatever is defined in the /etc/named.conf, as I have done.

In the plugin code, it says "where syslog puts daemon output - ie the named statistics output."

Simply put, in no file on my system is there XSTATS or NSTATS lines related to bind-9.11.24.

**Comment by sumpfralle** [comment]
How about `journalctl`?

**Comment by dwreski** [comment]
What about it? :-)

I've searched through all of journalctl output (and /var/log/messages) and there are no occurrences of XSTAT or NSTAT.

Perhaps this is from Solaris that it mentions in the plugin? Was this tested on fedora or with bind-9.11?

**Comment by sumpfralle** [comment]
Please take a look at the [documentation for bind](http://web.mit.edu/darwin/src/modules/bind/bind/doc/html/options.html). Your input is welcome!

**Comment by dwreski** [comment]
Are you asking for my help in trying to figure out how to get bind to produce the XSTATS and NSTATS data this plugin needs to work properly?

**Comment by niclan** [comment]
@dwreski : I was under the impression that you were the one asking for help to figure out things.  He is asking for your feedback when you've found out how to get your nice new bind to produce the required output since he can't be a expert in everything.

The lines the plugin looks for are expected in a general systemlog file, which on very recent Linux versions might be in `journalctl -u named` or in some specific file unde `/var/log`.  It sounds like you have exhausted all these posibilities. The named.stats file is certainly not in the right format for this plugin though I see it has enough stuff in there for 3 or 4 new plugins. Plugins not yet written.

Based on my cursory inspection of the docs and your findings on your system the named plugin will no longer work.

**Comment by dwreski** [comment]
Got it, thanks. I'm happy to help; I was just making sure I understood the request properly.

It definitely doesn't seem like this plugin is going to work on bind - whether it's Debian or Fedora, so I'm not even sure why it's being included. I gravitated towards the named.stats file initially because I'm familiar enough with bind to know none of that info is included in the regular system logs.

General searches for "bind xstats" kind of thing also shows nothing, so I think more development needs to be done.

For the time being, I'll continue to use the bind9 plugin which just utilizes the basic query info from the query logs.

**Comment by sumpfralle** [comment]
Indeed this munin-contrib repository is based on the collaboration of three parties:
* contributors: submitting new plugins or improvements of existing plugins
* repository maintainers (e.g. myself): reviewing submissions and maintaining infrastructure (CI)
* users: reporting issues and providing details on how to fix them

Thus your input is very welcome and essential for improving or fixing the plugins in this repository. Thanks!
